framework/aster-frame/src/trap/handler.rs

```diff
@@ -102,3 +125,69 @@ cpu_local! {
 pub fn in_interrupt_context() -> bool {
     IN_INTERRUPT_CONTEXT.load(Ordering::Acquire)
 }
+
+fn kernel_page_fault_handler(f: &TrapFrame) {
+    // We only create mapping: `vaddr = paddr + PHYS_OFFSET` in kernel page fault handler.
+    let page_fault_vaddr = x86_64::registers::control::Cr2::read().as_u64();
+    debug_assert!(is_kernel_vaddr(page_fault_vaddr as usize));
+
+    // Check kernel region
+    // FIXME: The modification to the offset mapping of the kernel code and data should not permitted.
+    let kernel_region = MemoryRegion::kernel();
+    debug_assert!(
+        page_fault_vaddr < kernel_region.base() as u64
+            || page_fault_vaddr > (kernel_region.base() + kernel_region.len()) as u64
+    );
+
+    // Check error code and construct flags
+    let error_code = PageFaultErrorCode::from_bits_truncate(f.error_code);
+    debug!(
+        "Handling kernel page fault. Page fault address:{:x?}; Error code:{:?}",
+        page_fault_vaddr, error_code
+    );
+    debug_assert!(!error_code.contains(PageFaultErrorCode::USER));
+    debug_assert!(!error_code.contains(PageFaultErrorCode::INSTRUCTION));
+    let mut flags = PageTableFlags::empty()
+        .set_present(true)
+        .set_executable(false);
+    if error_code.contains(PageFaultErrorCode::WRITE) {
+        flags = flags.set_writable(true);
+    }
+
+    // Handle page fault
+    let mut kernel_page_table = KERNEL_PAGE_TABLE.get().unwrap().lock_irq_disabled();
+    if error_code.contains(PageFaultErrorCode::PRESENT) {
+        // Safety: The page fault address has been checked and the flags is constructed based on error code.
+        unsafe {
+            kernel_page_table
+                .protect(page_fault_vaddr as usize, flags)
+                .unwrap();
+        }
+    } else {
+        // Safety: The page fault address has been checked and the flags is constructed based on error code.
+        let paddr = page_fault_vaddr as usize - PHYS_OFFSET;
+        unsafe {
+            kernel_page_table
+                .map(page_fault_vaddr as usize, paddr, flags)
+                .unwrap();
+        }
+
+        // If the new mapping add one entry to p4, then we need to add the entry to ALL_MAPPED_PTE and the new page table
+        // will benefit from the mapping.
+        //
+        // Safety: page_directory_base is read from kernel page table, the address is valid.
+        let p4 = unsafe { table_of::<PageTableEntry>(kernel_page_table.root_paddr()).unwrap() };
+        let pte_index = PageTableEntry::page_index(page_fault_vaddr as usize, 4);
+        let mut map_pte = ALL_MAPPED_PTE.lock();
+        map_pte.entry(pte_index).or_insert(p4[pte_index]);
+
+        // Although the mapping is constructed in the kernel page table, the mapping may not added to the current page table.
+        //
+        // Safety: The modification will not affect kernel safety and the virtual address doesn't belong to the user virtual address.
+        unsafe {
```

Comment:
Here is confused for me.  The comment say "the mapping **may** not added to the current page table" but the code directly add the it to current page table.  Is this because tlb flush is not called here? So when will it be updated？